src/routing/comm.rs

```diff
@@ -123,7 +141,7 @@ impl Comm {
 
     /// Tests whether the peer is reachable.
     pub async fn is_reachable(&self, peer: &SocketAddr) -> Result<(), SendError> {
-        self.endpoint
+        self.connectivity_endpoint
```

Comment:
just to confirm that this means we have to carry this `test only endpoint` along the whole life.
though qp2p/quinn may take care of the idle issue, still a waste for some underneath keep-alive resources.
shall we instead here create a temp endpoint to test ? i.e. discard after the test immediately ?

Reply:
Yeah that is the other option. Let's discuss this on Slack with the team.